elytron/src/main/java/org/wildfly/extension/elytron/AbstractCredentialStoreResourceDefinition.java

```diff
@@ -90,10 +100,53 @@ abstract class AbstractCredentialStoreResourceDefinition extends SimpleResourceD
             .setMinSize(1)
             .build();
 
+    static final SimpleAttributeDefinition SIZE = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.SIZE, ModelType.STRING, true)
+            .setMinSize(1)
+            .build();
+
+    static final SimpleAttributeDefinition ALGORITHM = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.ALGORITHM, ModelType.STRING, true)
+            .setMinSize(1)
+            .build();
+
     static final SimpleAttributeDefinition KEY = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.KEY, ModelType.STRING, false)
             .setMinSize(1)
             .build();
 
+    static final SimpleAttributeDefinition PUBLIC_KEY_STRING = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.PUBLIC_KEY_STRING, ModelType.STRING, true)
+            .setMinSize(1)
+            .build();
+
+    static final SimpleAttributeDefinition PRIVATE_KEY_STRING = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.PRIVATE_KEY_STRING, ModelType.STRING, true)
+            .setMinSize(1)
+            .build();
+
+    static final SimpleAttributeDefinition PUBLIC_KEY_LOCATION = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.PUBLIC_KEY_LOCATION, ModelType.STRING, true)
```

Comment:
Have you checked to verify that an error occurs if neither are specified?

Reply:
> Since either PUBLIC_KEY or PUBLIC_KEY_LOCATION should be specified, we should use `setAlternatives` to specify this. For some examples, there are other places in the Elytron subsystem where `setAlternatives` is used.

I have added `setAlternatives`, however, it's not working here how it is supposed to. I'm not sure if I'm not missing something.